app/src/main/java/com/example/trainingintive/presentation/BaseActivity.kt

```diff
@@ -0,0 +1,26 @@
+package com.example.trainingintive.presentation
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.example.trainingintive.di.ViewModelFactory
+import com.example.trainingintive.navigators.Navigator
+import javax.inject.Inject
+
+abstract class BaseActivity<N : Navigator> : AppCompatActivity() {
```

Comment:
Minor: Stick to the popular convention that generic types should be named T (or S, U, V, etc. - when you are using more generic types in one class). So in this case `T` instead of `N` :D